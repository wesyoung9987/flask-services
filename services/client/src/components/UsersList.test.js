import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UserList from './UsersList';

const users = [
  {
    'active': true,
    'email': 'test@gmail.com',
    'id': 1,
    'username': 'bill'
  },
  {
    'active': true,
    'email': 'ron@mail.org',
    'id': 2,
    'username': 'ron'
  }
];

test('UserList renders properly', () => {
  const wrapper = shallow(<UserList users={users}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('bill');
});

test('UserList renders a snapshot correctly', () => {
  const tree =  renderer.create(<UserList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
