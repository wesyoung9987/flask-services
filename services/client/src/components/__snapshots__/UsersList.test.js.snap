// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders a snapshot correctly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    bill
  </h4>
  <h4
    className="box title is-4"
  >
    ron
  </h4>
</div>
`;
